cmd/api-linter/cli.go

```diff
@@ -105,14 +111,33 @@ func runCLI(rules lint.Rules, configs lint.Configs, args []string) error {
 				case "json":
 					marshal = json.Marshal
 				}
-				b, err := marshal(problems)
+				b, err := marshal(lintResponses)
 				if err != nil {
 					return err
 				}
-
 				if _, err = w.Write(b); err != nil {
 					return err
 				}
+
+				if c.BoolT("summary") {
+					summary, longestRuleLen := createSummary(lintResponses)
```

Comment:
This is a lot of logic that could probably be completely encapsulated in summary_cli.go and not clutter cli.go.  

```
if c.BoolT("summary") {
  emitSummary(w, lintResponses)
}
```